### DISCUSSION TITLE
How to do localization translation for software

### DISCUSSION THREAD

**Discussion by snmin** [totumonline/totum-ru-issues-and-discussions#93] [Как это сделать?] [answered]
Hello, I want to do software localization translation, can you provide a translation package? I only found some translation documents in the "totum-mit/totum/common/Lang" directory. Are there any other documents that need to be translated?
Can these documents that need to be translated be stored in a directory?

**Comment by poluninatm** [comment, marked as answer]
Hello. Let's assume: the language is "your language"(yl). Localization documents should be stored in directories: 

totum/common/Lang/YL.php - php class with translated phrases and local functions
http/js/i18n/yl.js - frontend localization
totum/moduls/install/start_yl.json.gz.ttm - scheme with local names. To create this you can unpack the internal json of the Russified version of the schema and translate everything you find in Cyrillic.

If you don't update the php or js with the changes in our repository, the user will see the texts in English. If you don't change the schema, it might cause an error or you might not see some improvements.

**Reply by snmin** [reply]
Thanks for the reply, ok, I have finished translating part of it.

**Comment by snmin** [comment]
Hello, I have translated two documents and one document has not been translated. The problem currently faced is that the format of the document content is not uniform, and it is too tiring to translate, and it is necessary to rearrange the specific parts that need to be translated, especially if there is no basis for this project, it is difficult to make accurate translations.
I have an application where I would like to file all the documents that need to be translated in a directory and in a unified format, preferably with a complete English translation. This will allow more people to translate for the project, and will be more conducive to the promotion of the project. Of course, I know that it is very difficult, and some frameworks need to be modified.
Here are two documents I translated, I tested it by changing the filename to Russian.There is still a part of the zh.js file inside that I don't know how to translate, because the corresponding key was not found.
[zh_CN.zip](https://github.com/totumonline/totum-ru-issues-and-discussions/files/8379213/zh_CN.zip)

**Reply by ttm-alexeypolunin** [reply]
After little inner discussion we think next.

In next week we change the endpoint-templates for languages to this construction: Base-template + Lang-file = Endpoint-template for lang.

You will have readable lang-file for translation. When we will be expand base-template you will be need to expand Lang-file for you lang based on En lang-file.

**Comment by snmin** [comment]
I'm glad to hear from you, and glad you took my proposal seriously. Thank you! !
